src/main/java/LoginServlet.java

```diff
@@ -0,0 +1,35 @@
+import com.google.appengine.api.users.User;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.JsonObject;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    JsonObject jsonResponse = new JsonObject();
+    UserService userService = UserServiceFactory.getUserService();
+
+    if (!userService.isUserLoggedIn()) {
+      jsonResponse.addProperty("loggedIn", false);
+      // After login redirect user to nickname-input page
+      String loginUrl = userService.createLoginURL("/");
```

Comment:
I didn't get redirected to a nickname input page after I logged in, the homepage just refreshed.

Reply:
^^

Reply:
This comment was left over since I used basically the same code from my portfolio, there isn't a nickname input page, removed comment